include/boost/geometry/algorithms/for_each.hpp

```diff
@@ -326,9 +326,7 @@ struct for_each_polygon
             return false;
         }
 
-        typename interior_return_type<Polygon>::type
-            rings = interior_rings(poly);
-
+        auto&& rings = interior_rings(poly);
```

Comment:
Forwarding reference, are we planning to forward rings?

Reply:
Right, if it's non-const then we have to use `auto&&`. If we use `auto&` the lifetime of a temporary which might be returned by `interior_rings()` will not be extended.

Reply:
thanks